src/com/google/enterprise/adaptor/prebuilt/RegexDecisionFilter.java

```diff
@@ -0,0 +1,285 @@
+ * {@code key} metadata property matches the regular expression is determined.
+ * If {@code pattern} is not set, then whether any key named {@code key} is
+ * present determines the match. The key {@code decideOnMatch} determines
+ * whether the {@code decision} is made for matching documents (if that key is
+ * set to {@code true}) or made for non-matching documents (if that key is set
+ * to {@code false}). By default, both Document {@code Metadata} and
+ * {@code params} are searched for the matching {@code key}; the config key
+ * {@code corpora} may be set to {@code metadata} or to {@code params} to
+ * restrict the search to only {@code Metadata} or {@code params}, respectively.
+ * Most keys/values of interest will normally be specified in the document's
+ * {@code Metadata}, but some key/values of interest (e.g. ContentType, DocId)
+ * exist in the document's {@code params}.
+ *
+ * <p>Example: skip documents that have a {@code NoIndex} metadata key or params
+ * key, regardless of value:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.SkipDocumentFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=NoIndex
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index
+   </code></pre>
+ *
+ * <p>Example: drop the content of documents that have a {@code ContentLength}
+ * greater than or equal to 100 megabytes:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexDecisionFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=ContentLength
+   metadata.transform.pipeline.regexDecisionFilter.pattern=0*[1-9][0-9]{8,}
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index-content
+   </code></pre>
+ *
+ * <p>Example: skips documents whose Metadata {@code Classification} property
+ * is neither {@code PUBLIC} nor {@code DECLASSIFIED}:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexDecisionFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=Classification
+   metadata.transform.pipeline.regexDecisionFilter.pattern=(PUBLIC)|(DECLASSIFIED)
+   metadata.transform.pipeline.regexDecisionFilter.decideOnMatch=false
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index
+   metadata.transform.pipeline.regexDecisionFilter.corpora=metadata
+   </code></pre>
+ */
+public class RegexDecisionFilter implements MetadataTransform {
+  /**
+   * Which collections of keys/values to search.  Metadata, params, or both.
+   */
+  private enum Corpora {
+    METADATA("metadata"),
+    PARAMS("params"),
+    METADATA_OR_PARAMS("metadata or params");
```

Comment:
This enum's name doesn't match all the others (spaces instead of dashes). I think it was basically just intended to be used as a default, and never explicitly. Maybe we could either rename this to ANYWHERE("anywhere") or add multiple names for the lookup: METADATA_OR_PARAMS("metadata or params", "anywhere")?

Similarly, maybe we could add "index" as a valid synonym for "as-is", either here, or in TransmissionDecision itself.